File: banchus-common/src/main/java/pe/nanamochi/banchus/domain/storage/FileStorageProvider.java
package pe.nanamochi.banchus.domain.storage;

import java.io.IOException;
import java.nio.file.Path;
import java.util.List;
import java.util.Optional;

public interface FileStorageProvider {
  void initialize(List<String> buckets) throws IOException;

  Optional<byte[]> read(String bucket, String key);

  void write(String bucket, String key, byte[] content);

  void delete(String bucket, String key);

  boolean exists(String bucket, String key);

  Path getFileAsPath(String bucket, String key);
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/domain/storage/FileStorageProvider.kt
package pe.nanamochi.banchus.domain.storage

import pe.nanamochi.banchus.domain.enums.StorageBucket

interface FileStorageProvider {
    fun initialize(buckets: List<StorageBucket>)

    fun read(bucket: StorageBucket, key: String): ByteArray?

    fun write(bucket: StorageBucket, key: String, content: ByteArray)

    fun delete(bucket: StorageBucket, key: String)

    fun exists(bucket: StorageBucket, key: String): Boolean
}
